clang/lib/Driver/Driver.cpp

```diff
@@ -619,29 +620,20 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
     DeviceTripleStr =
         HostTriple.isArch64Bit() ? "nvptx64-nvidia-cuda" : "nvptx-nvidia-cuda";
     llvm::Triple CudaTriple(DeviceTripleStr);
-    // Use the CUDA and host triples as the key into the ToolChains map,
-    // because the device toolchain we create depends on both.
-    auto &CudaTC = ToolChains[CudaTriple.str() + "/" + HostTriple.str()];
-    if (!CudaTC) {
-      CudaTC = llvm::make_unique<toolchains::CudaToolChain>(
-          *this, CudaTriple, *HostTC, C.getInputArgs(), OFK);
-    }
-    C.addOffloadDeviceToolChain(CudaTC.get(), OFK);
+
+    auto CudaTC = &getOffloadingDeviceToolChain(C.getInputArgs(), CudaTriple,
```

Comment:
From my understanding and I might be wrong, the getOffloadingDeviceToolChain call passes back a reference to a ptr (unique_ptr), so this is just getting a pointer to that reference so it can be passed to addOffloadDeviceToolChain. I had to think about this for a bit while making the changes..

This is following a similar syntax to getToolChain right now, in the way it's used and implemented (can see line 691 in the patch for OMP using it).

Reply:
Ah yes, I got the precedence of operator wrong.
After checking https://en.cppreference.com/w/cpp/language/operator_precedence I understand.
I parsed the expression as `(&getOffloadingDeviceToolChain)()` and so I thought `getOffloadingDeviceToolChain()` because `&getOffloadingDeviceToolChain` and `getOffloadingDeviceToolChain` are the same for functions... Alzheimer...